pkg/k8s/api.go

```diff
@@ -109,6 +112,27 @@ func (kubeapi *kubernetesApi) checkApiAccess(client *http.Client) *healthcheckPb
 	return checkResult
 }
 
+func (kubeapi *kubernetesApi) checkDashboardAccess(client *http.Client) *healthcheckPb.CheckResult {
+	checkResult := &healthcheckPb.CheckResult{
+		Status:           healthcheckPb.CheckStatus_OK,
+		SubsystemName:    ConduitDashboardSubsystemName,
+		CheckDescription: ConduitDashboardCheckDescription,
+	}
+	dashboardEndpoint, err := kubeapi.UrlFor("conduit", "/services/web:http/proxy")
```

Comment:
This seems to repeat logic from [the command](https://github.com/runconduit/conduit/blob/97865bf11fa5371bd3bf93fb2c616cf2b79c0af9/cli/cmd/dashboard.go#L61). I am also not a big fan of adding anything dashboard-specific to this type, as it changes it from having one responsibility (knowing about k8s API) to two (knowing about the k8s API *and* about the Conduit dashboard).

Suggestion: create a package under `cli/dashboard` and a `Dashboard` object that not only abstracts this logic but also implements this `checkDashboardAccess` check. I am guessing that the `Dashboard` object will need an instance of `KubernetesApi`, but once it has that it can do all this stuff and it's a bit more cohesive.

Reply:
Good point. I agree that it breaks the single responsibility principle. I will go along with your suggestion.

Reply:
Taking a closer look at the code, it feels like the fundemental issue here is that we want to be able to see if the web endpoint is available. So that makes things think that all we need to do is to call the web endpoint first to see what response we get and then proceed to opening the  proxy. This stripped down version does a couple of things.
- foregoes the health check for the dashboard and rather just do a simple HTTP call to see if the web endpoint
- Improve k8s API cohesion by removing the "dashboard" context for `api.go`
- Reduce the level of indirection that could possibly be introduced by adding a new `Dashboard` object under `cli/dashboard`

Would love to hear your thoughts on that and whether the above makes sense?

Reply:
@deebo91 I like this idea a lot, I'd just `extract method` on the HTTP stuff into a `tryToReachDashboardViaHttp` func or something to keep the `RunE` body easier to read w/o implementation detail noise